### PR TITLE
Add model parameter for convective adjustment diffusivity

### PR DESCRIPTION
**Pull request by blallen**
# Description

Enable user to change the amount of convective adjustment for the HBModel at runtime by turning the convective adjustment diffusivity into a model parameter.

I have

- [x] Written and run all necessary tests with CLIMA by including `tests/runtests.jl`
- [x] Followed all necessary [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) and run `julia .dev/climaformat.jl .`
- [x] Updated the documentation to reflect changes from this PR.

# For review by CLIMA developers

- [x] There are no open pull requests for this already
- [x] CLIMA developers with relevant expertise have been assigned to review this submission
- [x] The code conforms to the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) and has consistent naming conventions. `julia .dev/format.jl` has been run in a separate commit.
- [x] This code does what it is technically intended to do (all numerics make sense physically and/or computationally)

### CODE REVIEW

**Comment by blallen** [comment]
bors try

File: src/Ocean/HydrostaticBoussinesq/HydrostaticBoussinesqModel.jl [hunk 1]
```diff
@@ -276,8 +279,11 @@ applies convective adjustment in the vertical, bump by 1000 if ∂θ∂z < 0
 - `∂θ∂z`: value of the derivative of temperature in the z-direction
 """
 @inline function diffusivity_tensor(m::HBModel, ∂θ∂z)
-    ∂θ∂z < 0 ? κ = (@SVector [m.κʰ, m.κʰ, 1000 * m.κᶻ]) : κ =
-        (@SVector [m.κʰ, m.κʰ, m.κᶻ])
+    ∂θ∂z < 0 ? κ = (@SVector [m.κʰ, m.κʰ, m.κᶜ]) : κ = (@SVector [
+        m.κʰ,
+        m.κʰ,
+        m.κᶻ,
+    ])
```
Comment (blallen) [inline src/Ocean/HydrostaticBoussinesq/HydrostaticBoussinesqModel.jl:286]: @simonbyrne @charleskawczynski any ideas on how I can trick the formatter into making this look nicer? the previous version is much easier to read

Comment (jkozdon) [reply src/Ocean/HydrostaticBoussinesq/HydrostaticBoussinesqModel.jl:286]: You can [turn off the formatter](https://domluna.github.io/JuliaFormatter.jl/stable/skipping_formatting/) for chunks for code. We do this in a [handful of places already](https://github.com/CliMA/ClimateMachine.jl/search?q=format%3A+off&unscoped_q=format%3A+off)

Comment (charleskawczynski) [reply src/Ocean/HydrostaticBoussinesq/HydrostaticBoussinesqModel.jl:286]: Or break it down into multiple lines with temporary arrays?

Comment (blallen) [reply src/Ocean/HydrostaticBoussinesq/HydrostaticBoussinesqModel.jl]: realized I only needed to change the third component of the array and I can just use a temporary variable for that and make the actual array/matrix in the `return` statement

**Comment by blallen** [comment]
bors try

**Pull request merged**